docs/distributions/ubuntu/upgrade.md

```diff
@@ -0,0 +1,55 @@
+# Upgrade
+
+Recommendations:
+
+* Perform upgrade from LTS to LTS.
+
+Warnings:
+
+* If you have important data and workspace, make sure that you have backup.
+* Make sure you have enough disk space.
+* **Ensure that you met all the below requirements so that you can recover in case of failed upgrade.**
+
+# Hardware Requirements
```

Comment:
Why Ethernet? Doesn't wifi work?

Reply:
No, `ifconfig -a` didn't show wifi interface. `iwconfig`, `ip link`, etc. I couldn't use `wpa_supplicant` or `iw` to connect.

Reply:
But I saw `brcmfmac` was loaded.

Reply:
What was `uname -r` after upgrading to 22.04

Reply:
That's kinda strange, cause after installing a custom kernel, technically there is no difference between a T2 Mac and a general PC, provided we load the kernel.

Is it possible that in your case the kernel was quite old?

Reply:
`uname -r` shows `5.14.16-t2-big-sur` after upgrade.
Seems Ubuntu 22.04 is shipped with 5.15 and mine was a little old, this is why I recommend users to upgrade their kernel to >= shipped version.
We can add using wifi in this doc, but I am not sure where we need to write it, and the exact instructions. Could someone else contribute to this?

Reply:
If you check the Kernel Requirements section, I've mentioned it.

Reply:
Then I don't think we need to specify Ethernet. In your case, the kernel booting wasn't correct.

Reply:
I put it in the Troubleshooting section, as additional info for users who might face similar problems (for any reason).